core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -190,6 +190,12 @@ class DAGScheduler(
   /**
    * Number of consecutive stage attempts allowed before a stage is aborted.
    */
+  private[scheduler] val unRegisterOutputOnHostOnFetchFailure =
+    sc.getConf.getBoolean("spark.fetch.failure.unRegister.output.on.host", true)
```

Comment:
@jiangxb1987 - Do we want to set the default to false? If we believe that this is the correct and expected behavior, we should set it to true and in case we see issues, we can turn it off?